### PR TITLE
fix(vep_parser): use nested schema for insilico predictors

### PR DESCRIPTION
**Pull request by project-defiant**
## ✨ Context

This PR closes [#3546](https://github.com/opentargets/issues/issues/3546). More context in the issue.

## 🛠 What does this PR implement

- [x] Addition of helper function to extract nested StructType.
- [x] Pass correct schema for `inSilicoPredictors` field.
- [x] Addition of data case variants with multiple (2) elements in `colocalised_variants` field in following combination - `[(rs123, None), (rs234, rs345)]`

## 🚦 Before submitting

- [x] Do these changes cover one single feature (one change at a time)?
- [x] Did you read the [contributor guideline](https://opentargets.github.io/gentropy/development/contributing/#contributing-checklist)?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you make sure there is no commented out code in this PR?
- [x] Did you follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standards in PR title and commit messages?
- [x] Did you make sure the branch is up-to-date with the `dev` branch?
- [x] Did you write any new necessary tests?
- [x] Did you make sure the changes pass local tests (`make test`)?
- [x] Did you make sure the changes pass pre-commit rules (e.g `poetry run pre-commit run --all-files`)?

### CODE REVIEW

File: src/gentropy/common/spark_helpers.py [hunk 1]
```diff
@@ -525,7 +542,7 @@ def get_value_from_row(row: Row, column: str) -> Any:
 
 
 def enforce_schema(
-    expected_schema: t.StructType,
+    expected_schema: t.ArrayType | t.StructType | Column | str,
```
Comment (DSuveges) [inline src/gentropy/common/spark_helpers.py:545]: Are you sure this function can deal with column and string types? This is where the schema is used:
```python
return f.from_json(f.to_json(function(*args, **kwargs)), expected_schema)
```

Essentially we are taking the return value of a function, cast it to json string, then reading it back from json following the schema provided.

File: src/gentropy/datasource/ensembl/vep_parser.py [hunk 2]
```diff
@@ -25,15 +26,19 @@
 
 
 class VariantEffectPredictorParser:
-    """Collection of methods to parse VEP output in json format."""
+    """Collection of methods to parse VEP output in json format.
+
+    NOTE: Due to the fact that the comparison of the xrefs is done om the base of rsids
+    if the field `colocalised_variants` have multiple rsids, this extracting xrefs will result in
+    an array of xref structs, rather then the struct itself.
```
Comment (DSuveges) [inline src/gentropy/datasource/ensembl/vep_parser.py:33]: I wouldn't add this comment as a class docstring. However certainly explain why the xref schema is an array.

File: src/gentropy/common/spark_helpers.py (hunk 1 above)
Comment (project-defiant) [reply src/gentropy/common/spark_helpers.py:545]: `to_json` can have any of these parameters - see https://downloads.apache.org/spark/docs/3.3.4/api/python/reference/pyspark.sql/api/pyspark.sql.functions.from_json.html?highlight=from_json

**Review by DSuveges — APPROVED** [review]
All looks OK, except this line: `expected_schema: t.ArrayType | t.StructType | Column | str,` See my comment, which is rather a question than a blocker.

File: src/gentropy/datasource/ensembl/vep_parser.py (hunk 2 above)
Comment (project-defiant) [reply src/gentropy/datasource/ensembl/vep_parser.py:33]: Fair enough

**Pull request merged**